### PR TITLE
feat(Scaler): Adds Solr Scaler

### PR DESCRIPTION
**Pull request by ithesadson**
_Provide a description of what has been changed_
+Adds Solr Scaler for autoscaling based on solr queries

### Checklist

- [x] When introducing a new scaler, I agree with the [scaling governance policy](https://github.com/kedacore/governance/blob/main/SCALERS.md)
- [x] I have verified that my change is according to the [deprecations & breaking changes policy](https://github.com/kedacore/governance/blob/main/DEPRECATIONS.md)
- [x] Tests have been added
- [x] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)
- [x] A PR is opened to update the documentation on ([repo](https://github.com/kedacore/keda-docs)) *(if applicable)*
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))

Relates to #4234 
Relates to https://github.com/kedacore/keda-docs/pull/1093

### CODE REVIEW

**Comment by ithesadson** [comment]
Hi @tomkerkhove , I wrote the scaler and it seems to be working.
If everything is ok and you approve, I can start writing the document.

**Comment by ithesadson** [comment]
@zroubalik Can you trigger the e2e test again, I increased the time given for scale up

**Comment by ithesadson** [comment]
I added the document for solr scaler. https://github.com/kedacore/keda-docs/pull/1093

File: CHANGELOG.md [hunk 1]
```diff
@@ -42,6 +42,7 @@ To learn more about active deprecations, we recommend checking [GitHub Discussio
 - [v1.0.0](#v100)
 
 ## Unreleased
+- **General:** Introduce new Solr Scaler ([#4234](https://github.com/kedacore/keda/issues/4234))
```
Comment (tomkerkhove) [inline CHANGELOG.md:45]: Nit - Can you move this to the "new" section please

File: pkg/scalers/solr_scaler.go [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+type solrScaler struct {
+	metricType v2.MetricTargetType
+	metadata   *solrMetadata
+	httpClient *http.Client
+	logger     logr.Logger
+}
+
+type solrMetadata struct {
+	host                       string
+	core                       string
```
Comment (tomkerkhove) [inline pkg/scalers/solr_scaler.go:27]: As per docs PR, wouldn't it be better to rename this to make it more end-user friendly?
```suggestion
	collection                 string
```

Comment (ithesadson) [reply pkg/scalers/solr_scaler.go:27]: Absolutely, I agree

**Pull request merged**